keps/sig-auth/4601-authorize-with-selectors/README.md

```diff
@@ -0,0 +1,875 @@
+//   a query is not something we want), see https://www.oxeye.io/resources/golang-parameter-smuggling-attack for more details.
+// For the kube-apiserver:
+// * If rawSelector is empty and requirements are empty, the request is not limited.
+// * If rawSelector is present and requirements are empty, the rawSelector will be parsed and limited if the parsing succeeds.
+// * If rawSelector is empty and requirements are present, the requirements should be honored
+// * If rawSelector is present and requirements are present, the request is invalid.
+type LabelSelectorAttributes struct {
+	// rawSelector is the serialization of a field selector that would be included in a query parameter.
+    // Webhook implementations are encouraged to ignore rawSelector.
+	// The kube-apiserver's SubjectAccessReview will parse the rawSelector. 
+	RawSelector string
+
+    // requirements is the parsed interpretation of a label selector.
+    // All requirements must be met for a resource instance to match the selector.
+    // Webhook implementations should handle requirements, but how to handle them is up to the webhook.
+    // Since requirements can only limit the request, it is safe to authorize as unlimited request if the requirements
+    // are not understood.
+	Requirements []metav1.LabelSelectorRequirement
+}
+
+type FieldSelectorRequirement struct {
+	// key is the label key that the selector applies to.
+	Key string `json:"key" protobuf:"bytes,1,opt,name=key"`
+	// operator represents a key's relationship to a set of values.
+	// Valid operators are In, NotIn, Exists, DoesNotExist
+	// The list of operators may grow in the future.
+	// Webhook authors are encouraged to ignore unrecognized operators and assume they don't limit the request.
+	// The semantics of "all requirements are AND'd will not change, so other requirements can continue to be enforced.
+	Operator LabelSelectorOperator `json:"operator" protobuf:"bytes,2,opt,name=operator,casttype=LabelSelectorOperator"`
+	// values is an array of string values. If the operator is In or NotIn,
+	// the values array must be non-empty. If the operator is Exists or DoesNotExist,
+	// the values array must be empty. This array is replaced during a strategic
+	// merge patch.
+	// +optional
+	// +listType=atomic
+	Values []string `json:"values,omitempty" protobuf:"bytes,3,rep,name=values"`
+}
+
+
+```
+
+Importantly, if old webhook authorizers do not honor these new field, they will assume the broadest possible access and fail closed.
+If old in-cluster authorization does not include field and label selectors, the kube-apiserver will assume the broadest possible access and fail closed.
+
+### Node Authorizer Changes
+
+The node authorizer will `node_authorizer.go` to check the `spec.nodeName` for List requests to ensure it matches the
+`nodeName` from its username (`system:node:<nodeName>`).
+Then we will remove the 
```

Comment:
We'll actually keep it in the bootstrap policy so clusters using clusters binding to system:nodes and not using NodeAuthorizer will continue to work, and just make get / list / watch pods requests ones the node authorizer authorizes using the graph and selectors rather than falling through to the bootstrap rules.